wp-includes/sqlite/class-wp-sqlite-translator.php

```diff
@@ -1547,17 +1547,64 @@ private function execute_describe() {
 
 	/**
 	 * Executes an UPDATE statement.
+	 * Supported syntax:
+	 *
+	 * UPDATE [LOW_PRIORITY] [IGNORE] table_reference
+	 *    SET assignment_list
+	 *    [WHERE where_condition]
+	 *    [ORDER BY ...]
+	 *    [LIMIT row_count]
+	 *
+	 * @see https://dev.mysql.com/doc/refman/8.0/en/update.html
 	 */
 	private function execute_update() {
 		$this->rewriter->consume(); // Update.
-
-		$params = array();
+		$limit    = $this->rewriter->peek(
+			array(
+				'type'  => WP_SQLite_Token::TYPE_KEYWORD,
+				'value' => 'LIMIT',
+			)
+		);
+		$order_by = $this->rewriter->peek(
+			array(
+				'type'  => WP_SQLite_Token::TYPE_KEYWORD,
+				'value' => 'ORDER BY',
+			)
+		);
+		$where    = $this->rewriter->peek(
+			array(
+				'type'  => WP_SQLite_Token::TYPE_KEYWORD,
+				'value' => 'WHERE',
+			)
+		);
+		$params   = array();
 		while ( true ) {
 			$token = $this->rewriter->peek();
 			if ( ! $token ) {
 				break;
 			}
 
+			/*
+			 * If the query contains a WHERE clause, and either a LIMIT or ORDER BY clause,
+			 * we need to rewrite the query to use a nested SELECT statement.
+			 * eg:
+			 * - UPDATE table SET column = value WHERE condition LIMIT 1;
+			 * will be rewritten to:
+			 * - UPDATE table SET column = value WHERE rowid IN (SELECT rowid FROM table WHERE condition LIMIT 1);
+			 */
+			if ( $token->value === 'WHERE' && ( $limit || $order_by ) ) {
+				$this->remember_last_reserved_keyword( $token );
+				$this->rewriter->consume();
+				$this->prepare_update_for_limit_or_order();
+			}
+			/*
+			* In case we rewrite the query, we need to skip the semicolon.
+			* This is because the semicolon becomes part of the nested SELECT statement, and it breaks the query.
+			*/
```

Comment:
```suggestion
			/*
			 * In case we rewrite the query, we need to skip the semicolon.
			 * This is because the semicolon becomes part of the nested SELECT statement, and it breaks the query.
			 */
```